Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +222,253 @@ function onInterval(): void {
+        let parsedVersion: ParsedVersion;
+        parsedVersion.major = parseInt(tokens[0]);
+        parsedVersion.minor = parseInt(tokens[1]);
+        parsedVersion.patch = parseInt(tokens[2]);
+        parsedVersion.suffix = tokens[3];
+        return parsedVersion;
+    }(tokens);
+
+    if (!parsedVersion.major || !parsedVersion.minor || !parsedVersion.patch) {
+        telemetry.logLanguageServerEvent('versionParsingFailure', { 'versionString': versionStr });
+    }
+
+    return parsedVersion;
+}
+
+function parsedVersionGreater(v1: ParsedVersion, v2: ParsedVersion): boolean {
+    // ParsedVersions cannot be compared if either have a suffix that is not 'insiders'
+    if ((v1.suffix && v1.suffix !== 'insiders') || (v2.suffix && v2.suffix !== 'insiders')) {
+        return false;
+    }
+
+    let diff: number = v2.major - v1.major;
+    if (diff) {
+        return diff > 0;
+    } else
+    if (diff = v2.minor - v1.minor) {
+        return diff > 0;
+    } else
+    if (diff = v2.patch - v1.patch) {
+        return diff > 0;
+    } else
+    if (!v2.suffix && v1.suffix === 'insiders') {
+        return true;
+    }
+    return false;
+}
+
+interface Asset {
+    name: string;
+    browserDownloadUrl: string;
+}
+
+interface Build {
+    name: string;
+    assets: Asset[];
+}
+
+function isAsset(input: any): input is Asset {
+    return input && input.name && typeof(input.name) === "string" && input.browser_download_url && input.browser_download_url && typeof(input.browser_download_url) === "string"; 
+}
+
+function isBuild(input: any): input is Build {
+    return input && input.name && typeof(input.name) === "string" && isArrayOfAssets(input.assets) && input.assets.length >= 4;
+}
+
+function isArrayOfAssets(input: any): input is Asset[] {
+    if (!(input instanceof Array)) {
+        return false;
+    }
+    for (const item of input) {
+        if (!isAsset(item)) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+function isReleaseJson(input: any): input is Build[] {
+    if (!input || !(input instanceof Array) || input.length === 0) {
+        return false;
+    }
+    return isBuild(input[0]);
+}
+
+async function installVsix(vsixLocation: string, updateChannel: string): Promise<void> {
+    // Get the path to the VSCode command -- replace logic later when VSCode allows calling of
+    // workbench.extensions.action.installVSIX from TypeScript w/o instead popping up a file dialog
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsCodeScriptPath: string | undefined = await async function(platformInfo): Promise<string | undefined> {
+        if (platformInfo.platform === 'win32') {
+            const vsCodeProcessPath: string = path.dirname(process.execPath);
+            return path.join(vsCodeProcessPath, 'bin', 'code.cmd');
+        } else {
+            const stdout: Buffer = await execSync("which code");
```

Comment:
This doesn't seem to account for if the user is using the insiders build.  I know the insiders version on Windows has a different name than the stable version and I would expect the mac/linux versions to have different names too.

Reply:
Also, I wouldn't expect `await execSync` to be doing what you expect.  Is there an async version of `exec` you can use instead?

Reply:
(there are additional `await execSync` lines that will likely need to be updated too)

Reply:
Yeah, `exec` itself is async. Can't remember why I chose `await execSync`. Doesn't make sense either in hindsight